zipkin/src/main/java/zipkin2/internal/UnsafeBuffer.java

```diff
@@ -98,19 +98,33 @@ public void reset() {
 
   String readUtf8(int length) {
     require(length);
-    String result = new String(buf, pos, length, UTF_8);
+    String result = maybeDecodeShortAsciiString(buf, pos, length);
+    if (result == null) result = new String(buf, pos, length, UTF_8);
     pos += length;
     return result;
   }
 
+  // Speculatively assume all 7-bit ASCII characters.. common in normal tags and names
+  @Nullable static String maybeDecodeShortAsciiString(byte[] buf, int offset, int length) {
```

Comment:
Did you happen to target this method specifically in the microbench? the `String(byte[], UTF_8)` constructor special-cases ASCII strings, similar to this code, and it might be even faster on JVM9+ where ascii byte arrays use `System.arrayCopy` without converting to chars at all.

Reply:
this method made the most difference, but no I didn't test it independently of decoding spans

Reply:
also my bench was JDK 1.8... anyway we can fine tune that no problem. the code is still drifting as I retrofit ByteBuffer again. We can revisit this as it is a very specific area and easy to change